model/task/task.go

```diff
@@ -2953,6 +2972,19 @@ func (t *Task) SetTaskGroupInfo() error {
 		}}))
 }
 
+func (t *Task) SetNumDependents() error {
+	if t.NumDependents == 0 {
```

Comment:
Would we never need to change dependencies from a higher number to a lower number?

Reply:
I don't believe so but I added handling for it